fe/fe-core/src/main/java/com/starrocks/catalog/Type.java

```diff
@@ -367,6 +378,7 @@ public abstract class Type implements Cloneable {
         compatibilityMatrix[DATETIME.ordinal()][DECIMAL32.ordinal()] = PrimitiveType.DECIMAL32;
         compatibilityMatrix[DATETIME.ordinal()][DECIMAL64.ordinal()] = PrimitiveType.DECIMAL64;
         compatibilityMatrix[DATETIME.ordinal()][DECIMAL128.ordinal()] = PrimitiveType.DECIMAL128;
+        compatibilityMatrix[DATETIME.ordinal()][UNKNOWN_TYPE.ordinal()] = PrimitiveType.INVALID_TYPE;
```

Comment:
why do we need to add the UNKNOWN_TYPE to compatibilityMatrix?

Reply:
> why do we need to add the UNKNOWN_TYPE to compatibilityMatrix?

Because `UNKNOWN_TYPE` is added to `PrimitiveType`, and types will be compared with `UNKNOWN_TYPE`.